Source/EasyNetQ/Logging/LoggerExtensions.cs

```diff
@@ -0,0 +1,523 @@
+        }
+
+        /// <summary>
+        ///     Check if the <see cref="LogLevel.Fatal"/> log level is enabled.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/> to check with.</param>
+        /// <returns>True if the log level is enabled; false otherwise.</returns>
+        public static bool IsFatalEnabled(this ILogger logger)
+        {
+            GuardAgainstNullLogger(logger);
+            return logger.Log(LogLevel.Fatal, null, null, EmptyParams);
+        }
+
+        /// <summary>
+        ///     Check if the <see cref="LogLevel.Info"/> log level is enabled.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/> to check with.</param>
+        /// <returns>True if the log level is enabled; false otherwise.</returns>
+        public static bool IsInfoEnabled(this ILogger logger)
+        {
+            GuardAgainstNullLogger(logger);
+            return logger.Log(LogLevel.Info, null, null, EmptyParams);
+        }
+
+        /// <summary>
+        ///     Check if the <see cref="LogLevel.Trace"/> log level is enabled.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/> to check with.</param>
+        /// <returns>True if the log level is enabled; false otherwise.</returns>
+        public static bool IsTraceEnabled(this ILogger logger)
+        {
+            GuardAgainstNullLogger(logger);
+            return logger.Log(LogLevel.Trace, null, null, EmptyParams);
+        }
+
+        /// <summary>
+        ///     Check if the <see cref="LogLevel.Warn"/> log level is enabled.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/> to check with.</param>
+        /// <returns>True if the log level is enabled; false otherwise.</returns>
+        public static bool IsWarnEnabled(this ILogger logger)
+        {
+            GuardAgainstNullLogger(logger);
+            return logger.Log(LogLevel.Warn, null, null, EmptyParams);
+        }
+
+        /// <summary>
+        ///     Logs a message at the <see cref="LogLevel.Debug"/> log level, if enabled.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/> to use.</param>
+        /// <param name="messageFunc">The message function.</param>
+        public static void Debug(this ILogger logger, Func<string> messageFunc)
+        {
+            GuardAgainstNullLogger(logger);
+            logger.Log(LogLevel.Debug, messageFunc, null, EmptyParams);
+        }
+
+        /// <summary>
+        ///     Logs a message at the <see cref="LogLevel.Debug"/> log level, if enabled.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/> to use.</param>
+        /// <param name="message">The message.</param>
+        public static void Debug(this ILogger logger, string message)
+        {
+            if (logger.IsDebugEnabled()) logger.Log(LogLevel.Debug, message.AsFunc(), null, EmptyParams);
```

Comment:
Why some methods have `if (logger.IsXXXEnabled())` and some do not ?

Reply:
Will check.

Reply:
I've added missing checks.